eclass/llvm-r2.eclass

```diff
@@ -0,0 +1,435 @@
+					;;
+				--ldflags)
+					do_echo "-L${libdir}"
+					;;
+				--libs)
+					# llvm-config returns LLVM-<ver> but we're passing -L
+					# so we should be fine (and avoid NNgit... versions)
+					do_echo "-lLLVM"
+					;;
+				--system-libs)
+					do_echo
+					;;
+				--shared-mode)
+					do_echo shared
+					;;
+				--has-rtti)
+					do_echo YES
+					;;
+				--link-shared)
+					;;
+				--bindir)
+					if [ -n "${bindir}" ]; then
+						do_echo "${bindir}"
+					else
+						do_echo "CBUILD LLVM not available" >&2
+						exit 1
+					fi
+					;;
+				--includedir)
+					do_echo "${includedir}"
+					;;
+				--libdir)
+					do_echo "${libdir}"
+					;;
+				-*)
+					do_echo "Unsupported option: \${arg}" >&2
+					exit 1
+					;;
+				*)
+					# ignore components, we always return the dylib
+					;;
+			esac
+		done
+	EOF
+}
+# @FUNCTION: llvm_cbuild_setup
+# @DESCRIPTION:
+# Prepend the PATH for selected LLVM version in CBUILD.
+#
+# This function is meant to be used when the package in question uses
+# LLVM tools at build time.  It is called automatically
+# by llvm-r2_pkg_setup if LLVM is found installed in BROOT.
+#
+# Note that llvm-config from this path must not be used to build against
+# LLVM, as that will break cross-compilation.
+llvm_cbuild_setup() {
+	debug-print-function ${FUNCNAME} "$@"
+
+	local broot_prefix=$(get_llvm_prefix -b)
+	einfo "Using ${broot_prefix} for CBUILD LLVM ${LLVM_SLOT}"
+	[[ -d ${broot_prefix}/bin ]] ||
+		die "LLVM ${LLVM_SLOT} not found installed in BROOT (expected: ${broot_prefix}/bin)"
+
+	llvm_fix_clang_version CC CPP CXX
+	# keep in sync with profiles/features/llvm/make.defaults!
+	llvm_fix_tool_path ADDR2LINE AR AS LD NM OBJCOPY OBJDUMP RANLIB
+	llvm_fix_tool_path READELF STRINGS STRIP
+
+	# Set LLVM_CONFIG to help Meson (bug #907965) but only do it
+	# for empty ESYSROOT (as a proxy for "are we cross-compiling?").
+	if [[ -z ${ESYSROOT} ]] ; then
+		llvm_fix_tool_path LLVM_CONFIG
+	fi
+
+	llvm_prepend_path -b "${LLVM_SLOT}"
+}
+
+# @FUNCTION: llvm_chost_setup
+# @DESCRIPTION:
+# Set the environment for finding selected LLVM slot installed
+# for CHOST.  Create llvm-config wrappers to satisfy legacy lookups.
+#
+# This function is meant to be used when the package in question uses
+# LLVM compiles against and links to LLVM.  It is called automatically
+# by llvm-r2_pkg_setup if LLVM is found installed in ESYSROOT.
+#
+# Note that the generated llvm-config may refer to CBUILD installation
+# of LLVM via --bindir, if it is found available.
+llvm_chost_setup() {
+	debug-print-function ${FUNCNAME} "$@"
+
+	local esysroot_prefix=$(get_llvm_prefix -d)
+	einfo "Using ${esysroot_prefix} for CHOST LLVM ${LLVM_SLOT}"
+	[[ -d ${esysroot_prefix} ]] ||
+		die "LLVM ${LLVM_SLOT} not found installed in ESYSROOT (expected: ${esysroot_prefix})"
+
+	# satisfies find_package() in CMake
+	export CMAKE_PREFIX_PATH="${sysroot_prefix}${CMAKE_PREFIX_PATH+:${CMAKE_PREFIX_PATH}}"
```

Comment:
As pointed out in #39749, this affects CC/CXX lookup as well, so setting it is wrong. My next best idea is setting the bunch of `LLVM_ROOT`, `Clang_ROOT` (and so on) variables for `find_package()`.